MSAL/src/MSALPublicClientApplication.m

```diff
@@ -1576,6 +1576,55 @@ - (void)getDeviceInformationWithParameters:(MSALParameters *)parameters
     [deviceInfoProvider deviceInfoWithRequestParameters:requestParams completionBlock:block];
 }
 
+- (void)getWPJMetaDataDeviceWithParameters:(nullable MSALParameters *)parameters
+                               forTenantId:(nullable NSString *)tenantId
+                           completionBlock:(nonnull MSALWPJMetaDataCompletionBlock)completionBlock
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Querying WPJ MetaData");
+    
+    __auto_type block = ^(MSALWPJMetaData * _Nullable wpjMetaData, NSError * _Nullable msidError)
+    {
+        NSError *msalError = nil;
+        
+        if (msidError)
+        {
+            msalError = [MSALErrorConverter msalErrorFromMsidError:msidError classifyErrors:YES msalOauth2Provider:self.msalOauth2Provider];
+        }
+        else
+        {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Retrieved metadata device info %@", MSID_PII_LOG_MASKABLE(wpjMetaData));
```

Comment:
Pass in requestParams here as context (second parameter)

Reply:
requestParams is not created until this point, similarly in getDeviceInformationWithParameters, we pass nil for context

Reply:
You can just move line 1615 up to line 1584 and have those created, right? It's a mistake it isn't done in the current code.